### PR TITLE
Fix %M to be formatted correctly when using printf

### LINKED ISSUE #1034: printf units of measure does not work for %M specifier
F# 4.0 has an improvement to [`printf` that supports units of measure](https://github.com/fsharp/FSharpLangDesign/blob/master/FSharp-4.0/PrintfUnitsOfMeasureDesignAndSpec.md). However it does not work for the `%M` specifier (for `System.Decimal`):

```fsharp
[<Measure>] type usd

printfn "%M" 1.m<usd>
```
Gives
> error FS0001: This expression was expected to have type
>     decimal    
> but here has type
>     decimal&lt;usd&gt;

Changing the `1.` literal to `float` and the format specifier to `%f` solves the issue:
```fsharp
[<Measure>] type usd

printfn "%f" 1.<usd>
```
Outputted:
> [&lt;Measure&gt;]
> type usd
> val a : string = "1.000000"

### PR DESCRIPTION
**Pull request by rexcfnghk**
This PR fixes #1034

### CODE REVIEW

File: src/fsharp/CheckFormatStrings.fs [hunk 1]
```diff
@@ -235,11 +235,7 @@ let parseFormatStringInternal (m:range) g (source: string option) fmt bty cty =
               | ('h' | 'H') ->
                   failwithf "%s" <| FSComp.SR.forHIsUnnecessary()
 
-              | 'M' -> 
-                  collectSpecifierLocation relLine relCol
-                  parseLoop ((posi, g.decimal_ty) :: acc) (i+1, relLine, relCol+1)
-
-              | ('f' | 'F' | 'e' | 'E' | 'g' | 'G') ->
+              | ('f' | 'F' | 'e' | 'E' | 'g' | 'G' | 'M') ->
```
Comment (dsyme) [inline src/fsharp/CheckFormatStrings.fs]: I think this is not what you want - ``let f x = printf "%M" x`` will now default to be of type ``float`` in the absence of other constraints.

**Comment by rexcfnghk** [comment]
Oops sorry, how can I fix that?

**Comment by rexcfnghk** [comment]
Do you mean I should keep the `M` pattern matching branch and change the `parseLoop` call to `parseLoop ((posi, mkFlexibleFormatTypar m [ g.decimal_ty ] g.decimal_ty`?

**Comment by dsyme** [comment]
Yes that's it.  

Ideally you could add a test to ensure that ``let f x = printf "%M" x`` gets decimal type, since that test is missing.  But I'm happy to accept the PR as is.

**Comment by rexcfnghk** [comment]
@dsyme Please let me know where should I add the test and I will add it. Thanks :)

**Comment by dsyme** [comment]
tests/fsharp/typecheck/sigs - add a new "pos" test with both an implementation and a signature file

**Comment by rexcfnghk** [comment]
@dsyme Is this what you want?

`pos23.fs`:
```fsharp
module PrintfDecimalUnitOfMeasureTests

[<Measure>] type usd

let f x = 
    let _ = sprintf "%M" x
    exit (match box x with :? decimal -> 0 | _ -> 1)
    
f 1.m<usd>
```

**Comment by dsyme** [comment]
Yes, this is fine.  

@KevinRansom  This PR is ready to go

**Pull request merged**